app/grandchallenge/components/form_fields.py

```diff
@@ -102,10 +105,27 @@ def get_initial_value(self):
             return self.initial
 
     def get_image_field(self):
+        current_value = None
+
+        if isinstance(self.initial, ComponentInterfaceValue):
+            current_value = self.initial.image
+        elif self.initial:
+            try:
+                uuid = UUID(self.initial, version=4)
+            except ValueError:
+                uuid = None
+
+            if uuid and Image.objects.filter(pk=uuid).exists():
+                current_value = Image.objects.get(pk=uuid)
+            elif uuid and UserUpload.objects.filter(pk=uuid).exists():
+                current_value = UserUpload.objects.get(pk=uuid)
+            else:
+                raise RuntimeError(f"Unknown image pk: {self.initial}")
```

Comment:
This is a bit misleading, since `self.initial` does not need to be an image pk. I would just say something along the lines of "Unknown type for initial value"

Reply:
fixed